### PR TITLE
[Merged by Bors] - Sub app label changes

### PR DESCRIPTION
**Pull request by cart**
Makes some tweaks to the SubApp labeling introduced in #2695:

* Ergonomics improvements
* Removes unnecessary allocation when retrieving subapp label
* Removes the newly added "app macros" crate in favor of bevy_derive
* renamed RenderSubApp to RenderApp

@zicklag (for reference)

### CODE REVIEW

File: crates/bevy_app/src/app.rs [hunk 1]
```diff
@@ -608,10 +607,16 @@ impl App {
         self
     }
 
-    pub fn sub_app_mut(&mut self, label: impl SubAppLabel) -> Option<&mut App> {
-        let label = Box::new(label) as BoxedSubAppLabel;
+    /// Retrieves a "sub app" stored inside this [App]. This will panic if the sub app does not exist.
+    pub fn sub_app(&mut self, label: impl AppLabel) -> &mut App {
+        self.get_sub_app(label)
+            .expect("SubApp with the given label does not exist")
```
Comment (alice-i-cecile) [inline crates/bevy_app/src/app.rs:613]: This should return the label's identity in the error message.

**Review by alice-i-cecile — APPROVED** [review]
Other than my error reporting nit, looks good to me.

File: crates/bevy_app/src/app.rs (hunk 1 above)
Comment (cart) [reply crates/bevy_app/src/app.rs]: Good call!

**Review by zicklag — APPROVED** [review]
Looks good to me! :+1:

**Pull request closed without merging**